pysmt/solvers/qelim.py

```diff
@@ -42,3 +48,39 @@ def __exit__(self, exc_type, exc_val, exc_tb):
         the associated resources.
         """
         del self
+
+
+class ShannonQuantifierEliminator(QuantifierEliminator, IdentityDagWalker):
+
+    LOGICS = [pysmt.logics.BOOL]
+
+    def __init__(self, environment, logic=None):
+        IdentityDagWalker.__init__(self, env=environment)
+        self.logic = logic
+
+    def eliminate_quantifiers(self, formula):
+        logic = get_logic(formula, self.env)
+        if not logic <= pysmt.logics.BOOL:
+            raise NotImplementedError("Shannon quantifier elimination only "\
```

Comment:
In principle yes, the problem is that this QE is complete for BOOL, but incomplete for any other theory (we have no way of distinguishing a formula having only boolean variables quantified).

The pysmt framework for QE does not fit well for this case: shall we declare this QE as supporting any logic and then throw an execption once a non-boolean variable is quantified?